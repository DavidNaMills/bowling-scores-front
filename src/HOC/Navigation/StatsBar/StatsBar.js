import React from 'react';
import classes from './StatsBar.module.scss';

const StatsBar = ({stats}) =>(
    <div className={classes.StatsBar}>
        <p>{`Average: ${stats.average}`}</p>
        <p>{`Games Played: ${stats.ttlGames}`}</p>
    </div>
)
export default React.memo(StatsBar);