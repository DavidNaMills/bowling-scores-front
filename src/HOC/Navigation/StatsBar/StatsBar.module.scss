@import '../../../styles/colors.scss';


.StatsBar {
    background-color: $btn_blue_bkg_active;
    // background-color: $btn_lightGreen_bkg;
    padding-left: 10px;
    padding-right: 10px;
    display: flex;
    justify-content: space-between;
    border-bottom: solid $btn_darkGreen_bkg_active 2px;
    font-size: 14px;
}