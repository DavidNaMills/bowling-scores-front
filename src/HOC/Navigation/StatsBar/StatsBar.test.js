import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import StatsBar from './StatsBar';


const stats = {
    average: 456,
    ttlGames: 11111
}

describe('<StatsBar /> component test suite', ()=>{
    it('snapshot test', ()=>{
        const comp = renderer.create(<StatsBar stats={stats}/>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });

    const wrapper = shallow(<StatsBar stats={stats}/>);

    it('component exists in the DOM', ()=>{
        expect(wrapper.exists()).toBeTruthy();
    });


    it('contains a <div> with StatsBar className', ()=>{
        const div = wrapper.find('div');
        expect(div.length).toBe(1);
        expect(div.props().className).toEqual('StatsBar');
    });
    
    it('<div> contains 2 <p> tags which contain the Average and Games Played', ()=>{
        const p = wrapper.find('div').find('p');
        expect(p.length).toBe(2);
        expect(p.at(0).text()).toEqual(`Average: ${stats.average}`);
        expect(p.at(1).text()).toEqual(`Games Played: ${stats.ttlGames}`);
    });
});
