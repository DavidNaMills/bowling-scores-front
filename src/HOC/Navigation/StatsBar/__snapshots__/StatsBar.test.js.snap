// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatsBar /> component test suite snapshot test 1`] = `
<div
  className="StatsBar"
>
  <p>
    Average: 456
  </p>
  <p>
    Games Played: 11111
  </p>
</div>
`;
